### PR TITLE
Fix ut: wait service ready after run Proxy server

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: longjiquan <jiquan.long@zilliz.com>
issue: #17201

### CODE REVIEW

File: internal/distributed/proxy/service_test.go [hunk 1]
```diff
@@ -18,8 +18,23 @@ package grpcproxy
...
 	"testing"
+	"time"
+
+	"google.golang.org/grpc/credentials"
+
+	"go.uber.org/zap"
+
+	"github.com/milvus-io/milvus/internal/log"
+
+	"google.golang.org/grpc"
```
Comment (aakejiang) [inline internal/distributed/proxy/service_test.go:37]: forma imports

Comment (longjiquan) [reply internal/distributed/proxy/service_test.go:32]: already formatted.

File: internal/distributed/proxy/service_test.go [hunk 2]
```diff
@@ -780,6 +795,105 @@ func (m *MockProxy) ListCredUsers(ctx context.Context, req *milvuspb.ListCredUse
...
 
+///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+type WaitOption struct {
+	Duration      time.Duration `json:"duration"`
+	Port          int           `json:"port"`
+	TLSEnabled    bool          `json:"tls_enabled"`
+	ClientPemPath string        `json:"client_pem_path"`
+	ClientKeyPath string        `json:"client_key_path"`
+	ClientCaPath  string        `json:"client_ca_path"`
```
[the remark this answers is not in the document]
Comment (longjiquan) [reply internal/distributed/proxy/service_test.go]: renamed.

File: internal/distributed/proxy/service_test.go [hunk 3]
```diff
@@ -780,6 +792,105 @@ func (m *MockProxy) ListCredUsers(ctx context.Context, req *milvuspb.ListCredUse
...
+
+// waitForGrpcReady block until service available or panic after times out.
+func waitForGrpcReady(opt *WaitOption) {
+	ticker := time.NewTicker(opt.Duration)
+	ch := make(chan error, 1)
+
+	go func() {
+		// just used in UT to self-check service is available.
+		address := "localhost:" + strconv.Itoa(opt.Port)
+		if !opt.TLSEnabled {
```
Comment (jaime0815) [inline internal/distributed/proxy/service_test.go:855]: Do we need to add a timeout for dial if Dial has not it?

Comment (longjiquan) [reply internal/distributed/proxy/service_test.go:855]: `grpc.WithTimeout` was deprecated by `grpc`. I add the timeout logic by setup a time ticker.

**Pull request merged**